File: src/main/java/ru/citeck/ecos/codeinsight/forms/components/Component.java
package ru.citeck.ecos.codeinsight.forms.components;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.ArrayList;
import java.util.List;

public abstract class Component {

    private String key = "";
    private List<String> refreshOn = new ArrayList<>();

    @JsonProperty("type")
    public abstract String getType();

    @JsonProperty("input")
    boolean getInput() {
        return false;
    }

    public String getKey() {
        return key;
    }

    public void setKey(String key) {
        this.key = key;
    }

    public List<String> getRefreshOn() {
        return refreshOn;
    }

    public void setRefreshOn(List<String> refreshOn) {
        this.refreshOn = refreshOn;
    }

}

File: src/main/kotlin/ru/citeck/idea/artifacts/action/codeinsight/forms/components/Component.kt
package ru.citeck.idea.artifacts.action.codeinsight.forms.components

import ecos.com.fasterxml.jackson210.annotation.JsonProperty

abstract class Component {

    private var label = ""
    private var key = ""
    private var refreshOn: List<String> = ArrayList()

    abstract fun getType(): String

    open fun setLabel(label: String) {
        this.label = label
    }

    open fun getLabel(): String {
        return label
    }

    open fun setKey(key: String) {
        this.key = key
    }

    open fun getKey(): String {
        return key
    }

    open fun setRefreshOn(refreshOn: List<String>) {
        this.refreshOn = refreshOn
    }

    open fun getRefreshOn(): List<String> {
        return refreshOn
    }

    @get:JsonProperty("input")
    open val input: Boolean
        get() = false
}
